docs/guides/using_link_props.rst

```diff
@@ -0,0 +1,123 @@
+.. _ref_guide_linkprops:
+
+=====================
+Using link properties
+=====================
+
+:index: property
+
+
+Links can contain **properties**. Due to how they're persisted under the hood,
+link properties have a few additional constraints: they're always *single* and
+*optional*.
+
+Declaration
+-----------
+
+.. code-block:: sdl
+
+  type Person {
+    required property name -> str { constraint exclusive; }
+
+    multi link friends -> Person {
+      property strength -> float64;
```

Comment:
IMO this would make all the following code samples look more complex. I added an explainer sentence at the top of the page explaining that `strength` corresponds to the strength of the friendshiop.